crates/matrix-sdk-ui/src/sync_service.rs

```diff
@@ -90,6 +111,58 @@ impl SyncTaskSupervisor {
         Self { task, termination_sender }
     }
 
+    async fn offline_check(
+        client: &Client,
+        receiver: &mut Receiver<TerminationReport>,
+    ) -> Option<TerminationReport> {
+        info!("Entering the offline mode");
+
+        let wait_for_termination_report = async {
+            loop {
+                // Since we didn't empty the channel when entering the offline mode in fear that
+                // we might miss a report with the
+                // `TerminationOrigin::Supervisor` origin and the channel might contain stale
+                // reports from one of the sync services, in case both of them have sent a
+                // report, let's ignore all reports we receive from the sync
+                // services.
+                let report =
+                    receiver.recv().await.unwrap_or_else(TerminationReport::supervisor_error);
+
+                match report.origin {
+                    TerminationOrigin::EncryptionSync | TerminationOrigin::RoomList => (),
```

Comment:
nit: `{}` instead of `()`

Reply:
5df0f2a3f3f5d3b78b65de6f738a03582ca11a31